lib/spack/spack/test/spec_semantics.py

```diff
@@ -41,23 +43,23 @@ def _specify(spec_like):
     return Spec(spec_like)
 
 
-def check_satisfies(target_spec, constraint_spec, target_concrete=False):
+def constrain_or_raise(target_spec, constraint_spec, target_concrete=False):
```

Comment:
Can this be called `check_can_constrain` instead?

Reply:
Refactored the tests, so this is outdated now.